src/prefect/cli/root.py

```diff
@@ -283,13 +283,24 @@ async def deploy(
     if project.get("pull") is None:
         project["pull"] = []
 
-    if not flow_name and not base_deploy["flow_name"] and not entrypoint:
+    if (
+        not flow_name
+        and not base_deploy.get("flow_name")
+        and not entrypoint
+        and not base_deploy.get("entrypoint")
+    ):
         exit_with_error("An entrypoint or flow name must be provided.")
-    if not name and not base_deploy["name"]:
+    if not name and not base_deploy.get("name"):
         exit_with_error("A deployment name must be provided.")
-    if (flow_name or base_deploy["flow_name"]) and entrypoint:
+    if (flow_name or base_deploy.get("flow_name")) and entrypoint:
```

Comment:
this might got unwieldy but I think we need to hedge against an entrypoint in the deployment.yaml spec as well:
```python
if (flow_name or base_deploy.get("flow_name")) and (entrypoint or base_deploy.get("entrypoint")):
```